packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/upgrade/upgrade.controller.js

```diff
@@ -0,0 +1,55 @@
+import find from 'lodash/find';
+
+import { MODES } from '../components/termsOfUse/terms-of-use.constants';
+import { BACKUP_NAME } from './upgrade.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($translate, dedicatedCloudDatacenterBackupService) {
+    this.$translate = $translate;
+    this.dedicatedCloudDatacenterBackupService = dedicatedCloudDatacenterBackupService;
+    this.MODES = MODES;
+    this.BACKUP_NAME = BACKUP_NAME;
+  }
+
+  $onInit() {
+    this.upgrading = false;
+    if (this.backup && this.backup.backupOffer) {
+      this.selectedOffer = find(this.backupOffers, {
+        offerName: this.backup.backupOffer,
+      });
+    }
+    this.conditionsAccepted = false;
+  }
+
+  upgrade() {
+    this.upgrading = true;
+    const capabilities = {
+      backupDurationInReport: this.backup.backupDurationInReport,
+      backupSizeInReport: this.backup.backupSizeInReport,
+      diskSizeInReport: this.backup.diskSizeInReport,
+      fullDayInReport: this.backup.fullDayInReport,
+      restorePointInReport: this.backup.restorePointInReport,
+      mailAddress: this.backup.mailAddress,
+      backupOffer: this.backup.backupOffer,
+    };
+    return this.dedicatedCloudDatacenterBackupService
+      .updateBackupCapabilities(this.productId, this.datacenterId, capabilities)
```

Comment:
Use `pick` as done in https://github.com/ovh/manager/pull/2320/files#diff-f1f66957f52691f22d275f3fcec7a002R48